sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_update(progress_bar_t *progress_bar)
```

Comment:
May I suggest a slight API addition?
If you factor out a function:

```
void progress_bar_print(const char *prefix, const char *suffix, uint8_t percent);
```

Most users can simply call that without the need to keep the progress_bar_t structure (saving 68bytes of RAM per instance).
That function can be called from `progress_bar_update()`, then using the struct. That way you keep the current API working as is.

Reply:
Good idea!